### DISCUSSION TITLE
Semester 4 Week 2

### DISCUSSION THREAD

**Discussion by Stev3eM** [SE-TINF22B6/CookHub#133] [Show and tell]
Hello backpackers,

This is our weekly discussion-blog to keep you up to date within our project progression.
A new week has approched and we got some new updates for you:

<br><br>

### Search Bar
✅ Implemented a search bar which shows all recipes from database and if there are already adventurized versions of recipe available.
![SearchBar](https://github.com/SE-TINF22B6/CookHub/assets/113697513/e597dfe1-7633-421d-ae02-c43e9a66485c)

### OpenAPI 
✅  API call inculding recipe and predefined prompt is working (answer will not be saved yet).
![image](https://github.com/SE-TINF22B6/CookHub/assets/77683850/e13c7d74-5fa8-4d2d-a51a-770b38ebf0a8)
The API endpoint takes the recipes ID as a parameter, fetches the recipe data from the database and generates the adventure text with a request to the OpenAI API. For more information on how we create these texts, visit our wiki entry about the [adventurize it](https://github.com/SE-TINF22B6/CookHub/wiki/adventurize-it-%E2%92%B8-%5B-v0.7-%5D) function.

### Bug Fixes
✅  Fixed several Bugs like:
- Lazy Loading errors in backend fixed (problems with serializing object attributes that are not initialized/loaded yet)
- Several UI bugs fixed (elements positions and sizes, additional scrollbars, general design)

<br><br>

See you next time!

**Your**   <img width="50" alt="image" src="https://github.com/SE-TINF22B6/CookHub/assets/113697513/1a218c37-d3b0-4c88-9ae1-827b5a4eba42">   **Team**

**Comment by fehlend0** [comment]
Great work, team! The addition of the search bar and OpenAPI integration are significant steps forward. It's reassuring to see the bug fixes, ensuring a smoother user experience. Keep up the momentum!
